src/index.mjs

```diff
@@ -150,14 +199,13 @@ export function listen(options) {
 * @return {Object} a Promise
 */
 export function prefetch(url, isPriority, conn) {
-  if (conn = navigator.connection) {
-    // Don't prefetch if using 2G or if Save-Data is enabled.
-    if (conn.saveData) {
-      return Promise.reject(new Error('Cannot prefetch, Save-Data is enabled'));
-    }
-    if (/2g/.test(conn.effectiveType)) {
-      return Promise.reject(new Error('Cannot prefetch, network conditions are poor'));
-    }
+  let chkConn = checkConnection(conn = navigator.connection);
```

Comment:
It's not clear why this is assigning to the `conn` parameter. Did you mean to set a default on line 201? Or just remove the `conn` parameter altogether? Or do `conn || navigator.connection`?

I guess this was already buggy...

Reply:
I have removed the `conn` parameter altogether as from my understanding it's not needed, but since it's been there in the earlier commits, perhaps @addyosmani can help to quickly double check to confirm it's ok, or is better to retain and use `conn || navigator.connection`? Thanks..